README.md

```diff
@@ -401,6 +401,19 @@ succeeding and which are failing.
 WIREIT_FAILURES=continue
 ```
 
+### Kill
+
+When a failure occurs in `kill` mode, running scripts are immediately
+terminated, and new scripts are not started. This mode is useful if you want to
+be notified as soon as possible about any failures. Since scripts may be
+terminated before they are finished, this mode risks leaving your output in a
```

Comment:
> This is true but written with fairly scary wording and by introducing the novel phrase "your output" it gives the reader the impression that this may cause difficult to predict problems. Then the next line "Wireit will recover from this state the next time it runs successfully" fills me with more uncertainty. How will it recover? What will it do?

I removed it. I think we just do a totally reasonable thing, and it would be weird to do anything else, so maybe it's not even worth the user's time. I also considered this phrasing which is framed in more reassuring terms:

```
While `kill` mode could cause a script to stop writing all of its output files
before it was finished, Wireit knows this and will always re-build or restore
from cache the next time it runs.
```

> My understanding as a Wireit power user and from skimming this PR: when we kill an in-flight process, it won't have finished writing its output files, and it may have written partial outputs. That's ok, because the next time the script runs it will get the chance to recover.

Yeah that's right. Because we always delete the `state` file before we start execution, and only write the `state` file or a cache entry if we succeed, then the next time you run, the script will 

> Implicit here is the assumption that it's ok to terminate a script at any time, and if it does incremental compilation then any incremental artifacts it uses (like .tsbuildinfo) can handle partial writes. I think this is a good assumption to make, processes are killed sometimes, that's part of life, and it's their responsibility to handle this gracefully.

Yeah, I agree that handling SIGINT gracefully is a normal thing that will not be a problem in the vast majority of cases (and it something a user is likely to do by themsleves.